pymrio/tools/iobalance.py

```diff
@@ -0,0 +1,189 @@
+
+
+    #error message for unbalanced sectors
+    class SectorsNotBalancedError(Exception):
+        def __init__(self, message="Sectors are not balanced (Sum of sector Inputs does not equal sum of Outputs)"):
+            self.message = message
+            super().__init__(self.message)  
+
+    #error message for incorrect total output
+    class TotalOutputError(Exception):
+        def __init__(self, message="Total Output does not Equal sum of inter-industry and final demand outputs"):
+            self.message = message
+            super().__init__(self.message)              
+              
+    
+    # Check shape of Inter industry is square
+    if isinstance(Inter_industry, pd.DataFrame):
+        num_rows_Z, num_cols_Z = Inter_industry.shape
+        if num_rows_Z == num_cols_Z:
+            pass
+        else:
+            raise NotSquareDataFrameError("Number of rows does not equal columns")
+    else:
+        pass
+    
+    #Check shape of other matrices
+    
+    #Value Added
+    if isinstance(value_added, pd.DataFrame):
+        num_rows_VA, num_cols_VA = value_added.shape
+        if num_cols_VA == num_cols_Z:
+            pass
+        else:
+            raise MatrixNotCompatibleError("Value Added shape is not compatible with InterIndustry Matrix")
+    else:
+        pass    
+    
+    #Check Final Demand
+    if isinstance(final_demand, pd.DataFrame):
+        num_rows_Y, num_cols_Y = final_demand.shape
+        if num_rows_Y == num_rows_Z:
+            pass
+        else:
+            raise MatrixNotCompatibleError("Final Demand shape is not compatible with InterIndustry Matrix")
+    else:
+        pass    
+        
+    
+    #Check Total Output
+    if isinstance(total_output, pd.DataFrame):
+        num_rows_X, num_cols_X = total_output.shape
+        if num_rows_X == num_rows_Z:
+            pass
+        else:
+            raise MatrixNotCompatibleError("Total Output shape is not compatible with InterIndustry Matrix")
+    else:
+        pass
+
+    print("Matrices Shapes are compatible")
+    
+    
+    #check if total Output is the sum of Industry and final demand outputs
+    for i in range(len(Inter_industry)):
+        
+        
+        Inter_industry_sum = Inter_industry.iloc[i].sum()
+        Final_demand_sum = final_demand.iloc[i].sum() 
+        total= Inter_industry_sum+Final_demand_sum
+        
+        
+        if round(total) == round(float(total_output.iloc[i])):
+            pass
+        else:
+            raise TotalOutputError('Total output does not equal sum of intermediate and final demand consumption')
+    
+    #check if system totals are balanced by summing total output and value added + InterIndustry
+    
+    X_sum = total_output.sum(axis=0) #Total Output sum
+    Z_sum_vertical = Inter_industry.sum(axis=0).sum() #Input Matrix sum (vertical sum of InterIndustry)
+    VA_sum = value_added.sum(axis=0).sum() #Value Added sum
+    
+    Input = Z_sum_vertical + VA_sum
```

Comment:
please follow python naming conventions (https://peps.python.org/pep-0008/#function-and-variable-names) except for the naming defined previously (Z, A, etc).